lib/index.js

```diff
@@ -38,6 +38,7 @@ internals.schema = Joi.object({
         protocol: Joi.string().valid('oauth', 'oauth2'),
         temporary: Joi.string().when('protocol', { is: 'oauth', then: Joi.required(), otherwise: Joi.forbidden() }),
         auth: Joi.string().required(),
+        authMethod: Joi.string().valid('basic', 'param').when('protocol', { is: 'oauth2', then: Joi.required(), otherwise: Joi.forbidden() }),
```

Comment:
I would implement this as a boolean since there are only two ways to authenticate. I would make the default using the header because that's the proper way to do it and allow for a params override.